pkg/machinepool/machinepool.go

```diff
@@ -19,11 +34,43 @@ import (
 var fetchMessage string = "Fetching %s '%s' for cluster '%s'"
 var notFoundMessage string = "Machine pool '%s' not found"
 
+// Regular expression to used to make sure that the identifier given by the
+// user is safe and that it there is no risk of SQL injection:
+var machinePoolKeyRE = regexp.MustCompile(`^[a-z]([-a-z0-9]*[a-z0-9])?$`)
+
 //go:generate mockgen -source=machinepool.go -package=machinepool -destination=machinepool_mock.go
 type MachinePoolService interface {
 	DescribeMachinePool(r *rosa.Runtime, cluster *cmv1.Cluster, clusterKey string, machinePoolId string) error
 	ListMachinePools(r *rosa.Runtime, clusterKey string, cluster *cmv1.Cluster) error
 	DeleteMachinePool(r *rosa.Runtime, machinePoolId string, clusterKey string, cluster *cmv1.Cluster) error
+	AddMachinePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r *rosa.Runtime,
+		args *MachinePoolArgs) error
+	AddNodePool(cmd *cobra.Command, clusterKey string, cluster *cmv1.Cluster, r *rosa.Runtime,
+		args *MachinePoolArgs) error
+}
+
+type MachinePoolArgs struct {
```

Comment:
NP : move this to a different file. This file is going to get very large and we can easily move this to its own file